front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```diff
@@ -478,28 +610,37 @@ const useTrackOccupancy = ({
 
     // Load zones for added trains, for each path operational point that has already been toggled at least once:
     if (addedTrainIDs.size || modifiedTrainIDs.size) {
-      forEach(pathOperationalPointsState, async (_, waypointId) => {
-        const newZones = await fetchTrackOccupancy(
+      forEach(pathOperationalPointsState, (opState, waypointId) => {
+        if (!opState?.selected) return;
```

Comment:
:question: Hum why do you do that ? I don't really get how opState works :thinking: